client/jetpack-cloud/sections/partner-portal/controller.tsx

```diff
@@ -231,3 +232,31 @@ export function requireSelectedPartnerKeyContext(
 		)
 	);
 }
+
+/**
+ * Require the user to have a valid payment method registered.
+ *
+ * @param {PageJS.Context} context PageJS context.
+ * @param {() => void} next Next context callback.
+ */
+export function requireValidPaymentMethod( context: PageJS.Context, next: () => void ) {
```

Comment:
We should only apply this redirect for agency partners or we'd prevent a12s from assigning licenses - there's an `isAgencyUser()` selector that was added recently we can use here.

Reply:
Updated! Now using `isAgencyUser()` selector for tracking if the user is of type agency.